illumos-utils/src/zfs.rs

```diff
@@ -254,19 +249,21 @@ pub fn get_all_omicron_datasets_for_delete() -> anyhow::Result<Vec<String>> {
     // This includes cockroachdb, clickhouse, and crucible datasets.
     let zpools = crate::zpool::Zpool::list()?;
     for pool in &zpools {
-        // For now, avoid erasing any datasets which exist on internal zpools.
-        if pool.kind() == crate::zpool::ZpoolKind::Internal {
-            continue;
-        }
+        let internal = pool.kind() == crate::zpool::ZpoolKind::Internal;
```

Comment:
Although this PR doesn't actually migrate any config data out of the ramdisk, eventually we'll move things from `/var/oxide` into `/pool/int/<UUID>/config` on the M.2s.

As a result, we *will* want to destroy internal datasets to "uninstall".